apis/v1alpha2/shared_types.go

```diff
@@ -249,6 +249,8 @@ type RouteParentStatus struct {
 	// The format of this field is DOMAIN "/" PATH, where DOMAIN and PATH are
 	// valid Kubernetes names
 	// (https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names).
+	//
+	// This field cannot be empty.
```

Comment:
In general we've only added explicit comments about a field being optional, but otherwise it is assumed to be required. With that said, I think we do need some more clarification here, so maybe something like this that also helps explain the purpose of the field?

```suggestion
	// Controllers MUST populate this field when writing status. Controllers should ensure that
	// entries to status populated with their ControllerName are cleaned up when they are no
	// longer necessary.
```

Reply:
Good call, added!